src/LanguageServer/ProtocolUnitTests/Diagnostics/AbstractPullDiagnosticTestsBase.cs

```diff
@@ -377,10 +377,7 @@ private sealed class MockTypescriptDiagnosticAnalyzer : DocumentDiagnosticAnalyz
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => [Descriptor];
 
-        public override Task<ImmutableArray<Diagnostic>> AnalyzeSemanticsAsync(Document document, CancellationToken cancellationToken)
-            => SpecializedTasks.EmptyImmutableArray<Diagnostic>();
-
-        public override Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(Document document, CancellationToken cancellationToken)
+        public override Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(TextDocument document, SyntaxTree? tree, CancellationToken cancellationToken)
```

Comment:
>TextDocument [](http://example.com/codeflow?start=76&length=12)

no need to check for it being a Document?

Reply:
No. This is just text code